### PR TITLE
Question options errors bag

### PR DESCRIPTION
**Pull request by Joaoviana**

### CODE REVIEW

File: client/src/pages/CreateSurvey/Options/Options.jsx [hunk 1]
```diff
@@ -35,17 +60,34 @@ const Option = ({ optionIndex, questionIndex }) => {
...
+      payload: [
+        `${questionIndex}${optionIndex}`,
+        `${questionIndex}${optionIndex + 1}`,
+      ],
+    });
   };
 
   const isNotFirstQuestion = optionIndex !== 0;
   const isNotLastQuestion = options.length - 1 !== optionIndex;
+  const optionTextIsBiggerThanMaxValue = text && text.length > 280;
```
Comment (TKostrzewski) [inline client/src/pages/CreateSurvey/Options/Options.jsx]: This variable name is being used above also, but within a function definition.

File: client/src/store/reducer/errorsBagReducer.js [hunk 2]
```diff
@@ -1,12 +1,37 @@
 const initialState = {};
 
+const swapKeyValuesInObject = (obj, key1, key2) => {
+  console.log('KEY1', key1);
```
Comment (TKostrzewski) [inline client/src/store/reducer/errorsBagReducer.js]: console logs

File: client/src/pages/CreateSurvey/Options/Options.jsx (hunk 1 above)
Comment (Joaoviana) [reply client/src/pages/CreateSurvey/Options/Options.jsx]: Yea true, but this one listens to the value `text` but the other one listens for the events value

File: client/src/store/reducer/errorsBagReducer.js (hunk 2 above)
Comment (Joaoviana) [reply client/src/store/reducer/errorsBagReducer.js]: removed in latest commit

**Comment by TKostrzewski** [comment]
When editing, if the survey title is empty, I get a helpful error message "Survey Title is required." but it is grey and not as obvious as the following. As soon as I type a letter in, I get a red error message.

File: client/src/store/reducer/errorsBagReducer.js [hunk 3]
```diff
@@ -1,12 +1,38 @@
 const initialState = {};
 
+const swapKeyValuesInObject = (obj, key1, key2) => {
+  const objectToSwapValues = {};
+  [objectToSwapValues[key1], objectToSwapValues[key2]] = [
+    objectToSwapValues[key2],
+    objectToSwapValues[key1],
```
Comment (AntonioGargaro) [inline client/src/store/reducer/errorsBagReducer.js]: I believe dot notation would look better here!

File: client/src/pages/CreateSurvey/Questions/Question.jsx [hunk 4]
```diff
@@ -92,16 +138,34 @@ const Question = ({ questionIndex }) => {
...
+                (e) =>
+                  setQuestionData(
+                    e,
+                    'text',
+                    'title',
+                    title,
+                    2,
+                    280,
+                    `question${questionIndex}`,
+                  )
```
Comment (AntonioGargaro) [inline client/src/pages/CreateSurvey/Questions/Question.jsx]: This inline function is quite hard too read easily, maybe making it a const would help?

File: client/src/pages/CreateSurvey/Questions/Question.jsx [hunk 5]
```diff
@@ -23,11 +23,42 @@ const Question = ({ questionIndex }) => {
...
+    if (inputType === 'text') {
+      const questionTitleIsBiggerThanMaxValue =
+        event.target.value && event.target.value.length > max;
+      const questionTitleIsSmallerThanMinValue =
+        event.target.value && event.target.value.length < min;
+      const questionTitleIsEmpty = event.target.value === '';
+      if (
+        questionTitleIsBiggerThanMaxValue ||
+        questionTitleIsSmallerThanMinValue ||
+        questionTitleIsEmpty
```
Comment (AntonioGargaro) [inline client/src/pages/CreateSurvey/Questions/Question.jsx]: You use this conditional quite a few times throughout this file and it spans multiple lines. Maybe you should write it as one single conditional that can be re-used everywhere.

File: client/src/pages/CreateSurvey/Options/Options.jsx [hunk 6]
```diff
@@ -13,11 +13,32 @@ import { ArrowUpward, ArrowDownward } from '@material-ui/icons';
...
+  const setOptionText = (event, optionTextIdentifier) => {
+    const optionTextIsBiggerThanMaxValue =
+      event.target.value && event.target.value.length > 280;
+    const optionTextIsSmallerThanMinValue =
+      event.target.value && event.target.value.length < 2;
+    const optionTextIsEmpty = event.target.value === '';
+    if (
+      optionTextIsBiggerThanMaxValue ||
+      optionTextIsSmallerThanMinValue ||
+      optionTextIsEmpty
```
Comment (AntonioGargaro) [inline client/src/pages/CreateSurvey/Options/Options.jsx]: See comment lower on this conditonal

File: client/src/store/reducer/errorsBagReducer.js (hunk 3 above)
Comment (marthaklambert) [reply client/src/store/reducer/errorsBagReducer.js]: Dot notation won't work here right? if theyre variables that are passed in

Comment (Joaoviana) [reply client/src/store/reducer/errorsBagReducer.js]: key1,key2 are strings. Would only work this way

**Pull request merged**